### PR TITLE
Added development structure, reversed order to release features

### PR DESCRIPTION
**Pull request by nkoenig**
Signed-off-by: Nate Koenig <nate@openrobotics.org>

# 🎉 New feature

## Summary

Adds a `Development` section to the main `index.yaml` file. Reverses the chronological order of the `release-features`, and fixed a rendering bug in one of the headers.

## Test it

Try it on https://staging.gazebosim.org.

## Checklist
- [ ] Signed all commits for DCO
- [ ] Added tests
- [ ] Added example and/or tutorial
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if the library has them)
- [ ] `codecheck` passed (See [contributing](https://gazebosim.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://gazebosim.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Agazebosim+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

File: development.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# Gazebo Development
+
+The following guides describe how Gazebo is developed, maintained, and
+released.
+
+* [Continuous Integration](ci)
+* [Contributing](contributing)
```
Comment (chapulina) [inline development.md:7]: It looks like the list is in alphabetical order, but for a contributor most likely they will look at the `contributing` page first.

File: index.yaml [hunk 2]
```diff
@@ -31,25 +31,28 @@ pages:
...
+      - name: release
+        title: Release process
+        file: release.md
+        description: Release process overview.
   - name: architecture
-    title: Architecture
+    title: Sim Architecture
     file: architecture.md
     description: Gazebo Sim's architecture overview.
     unlisted: true
```
Comment (chapulina) [inline index.yaml:55]: Do we want to get rid of the `unlisted`?

**Review by chapulina — APPROVED** [review]

File: index.yaml (hunk 2 above)
Comment (nkoenig) [reply index.yaml]: removed.

File: development.md (hunk 1 above)
Comment (nkoenig) [reply development.md:7]: reordered

**Pull request merged**